File: WordsmithHub.Domain/Address.cs
namespace WordsmithHub.Domain;

public abstract class Address
{
    public required string StreetInfo { get; set; }
    public string? AddressComplement { get; set; }
    public required string PostCode { get; set; }
    public required string City { get; set; }
    public string? State { get; set; }
    public required int CountryId { get; set; }
}

File: src/gui/src/models/Address.ts
import * as zod from "zod";

export type Address = {
    streetInfo: string;
    addressComplement: string | null;
    postCode: string;
    city: string;
    state: string | null;
    countryId: number;

}

export const addressSchema = zod
    .object({
        streetInfo: zod
            .string()
            .min(5, "Veuillez saisir une adresse valide"),
        addressComplement: zod
            .string()
            .nullable(),
        postCode: zod
            .string()
            .trim()
            .min(1, "Code postal requis")
            .max(5, "Code postal invalide"),
        city: zod
            .string()
            .trim()
            .min(1, "Nom de ville requis"),
        state: zod
            .string()
            .nullable(),
        countryId: zod
            .number()
    })